src/renderer/domains/collectives/model/referendum/model.ts

```diff
@@ -72,9 +72,42 @@ const {
   },
 });
 
+type ReferendumRequestParams = {
+  api: ApiPromise;
+  palletType: CollectivePalletsType;
+  chainId: ChainId;
+  referendums: ReferendumId[];
+};
+
+const { request } = createDataSource<CollectivesStruct<Referendum[]>, ReferendumRequestParams, Referendum[]>({
+  initial: $list,
+  async fn({ api, palletType, referendums }) {
+    const response = await referendaPallet.storage.referendumInfoFor(palletType, api, referendums);
+    const value: Referendum[] = [];
+    for (const { id, info } of response) {
+      if (!info) continue;
+      value.push(mapReferendum(id, info));
+    }
+
+    return value;
+  },
+  map(store, { params, result }) {
+    const currentStore = pickNestedValue(store, params.palletType, params.chainId) ?? [];
+    const updatedField = merge(
+      currentStore,
+      result,
+      x => x.id,
+      (a, b) => b.id - a.id,
+    );
+
+    return setNestedValue(store, params.palletType, params.chainId, updatedField);
+  },
+});
```

Comment:
this can be a bit hard to read but I am not sure if it's worth it to define functions separately

Reply:
moving map function away from declaration leads to additional types declarations and it looks worse. I've reduced code inside `fn`, it should help a bit